docs/migration_to_1.0.0.md

```diff
@@ -0,0 +1,326 @@
+natural: we only use imports and exports to read from the memory and write to it.
+
+Take a look at the following examples to get more details:
+* [Interacting with memory][memory]
+* [Using memory pointers][memory-pointers]
+
+The other thing where `vm::Ctx` was useful was to pass data from and to host functions. This has also been made simpler
+with Wasmer 1.x:
+
+```rust
+let shared_counter: Arc<RefCell<i32>> = Arc::new(RefCell::new(0));
+
+struct Env {
+    counter: Arc<RefCell<i32>>,
+}
+
+fn get_counter(env: &mut Env) -> i32 {
+    *env.counter.borrow()
+}
+
+let get_counter_func = Function::new_native_with_env(
+    &store, 
+    Env { counter: shared_counter.clone() }, 
+    get_counter
+);
+```
+
+A dedicated example describes how to use this feature: [Exposing host functions][host-functions].
+
+### Error handling
+
+Handling errors with Wasmer 0.x was a bit hard, especially, the `wasmer_runtime::error::RuntimeError`. It was rather 
+complex: it had many variants that you had to handle when pattern matching results. This has been made way simpler with 
+Wasmer 1.0.0:
+
+```diff
+// Retrieve the `get` function from module's exports and then call it
+let result = get.call(0, 13);
+
+match result {
+-     Err(RuntimeError::InvokeError(InvokeError::TrapCode { .. })) => {
+-         // ...
+-     }
+-     Err(RuntimeError::InvokeError(InvokeError::FailedWithNoError)) => {
+-         // ...
+-     }
+-     Err(RuntimeError::InvokeError(InvokeError::UnknownTrap { .. })) => {
+-         // ...
+-     }
+-     Err(RuntimeError::InvokeError(InvokeError::UnknownTrapCode { .. })) => {
+-         // ...
+-     }
+-     Err(RuntimeError::InvokeError(InvokeError::EarlyTrap(_))) => {
+-         // ...
+-     }
+-     Err(RuntimeError::InvokeError(InvokeError::Breakpoint(_))) => {
+-         // ...
+-     }
+-     Err(RuntimeError::Metering(_)) => {
+-         // ...
+-     }
+-     Err(RuntimeError::InstanceImage(_)) => {
+-         // ...
+-     }
+-     Err(RuntimeError::User(_)) => {
+-         // ...
+-     }
++     Error(e) => {
++         println!("Error caught from `div_by_zero`: {}", e.message());
++          
++         let frames = e.trace();
++         let frames_len = frames.len();
++         
++         // ...
++     }
+    Ok(_) => {
+        // ...
+    },
+}
+``` 
+
+As you can see here, handling errors is really easy now! You may find the following examples useful to get more familiar
+with this topic:
+* [Handling Errors][errors]
+* [Interrupting Execution][exit-early]
+
+Note than with Wasmer 1.0.0, each function that is part of the API has its own kind of error. For example:
+* Instantiating a module may return `InstantiationError`s;
+* Getting exports from the guest module may return `ExportError`s;
+* Calling a exported function may return `RuntimeError`s;
+* ...
+
+### Caching modules
+
+TODO: link to example, etc.
+
+### Metering
+
+TODO: link to example, etc.
```

Comment:
We don't strictly need this for merging the PR

Reply:
Indeed, I think we should just remove this part and wait until Metering is finished and then write the doc.

Reply:
Sounds good!